subxt/src/events/events_client.rs

```diff
@@ -37,6 +37,32 @@ where
     /// runtime upgrade. You can attempt to retrieve events from older blocks,
     /// but may run into errors attempting to work with them.
     pub fn at(
+        &self,
+        block_hash: T::Hash,
+    ) -> impl Future<Output = Result<Events<T>, Error>> + Send + 'static {
+        self.at_optional_block_hash(Some(block_hash))
+    }
+
+    /// Obtain events at the latest block hash.
+    ///
+    /// # Warning
+    ///
+    /// This call only supports blocks produced since the most recent
+    /// runtime upgrade. You can attempt to retrieve events from older blocks,
+    /// but may run into errors attempting to work with them.
+    pub fn at_latest(&self) -> impl Future<Output = Result<Events<T>, Error>> + Send + 'static {
+        self.at_optional_block_hash(None)
+    }
+
+    /// Obtain events at some block hash.
+    ///
+    /// # Warning
+    ///
+    /// This call only supports blocks produced since the most recent
+    /// runtime upgrade. You can attempt to retrieve events from older blocks,
+    /// but may run into errors attempting to work with them.
+    #[inline]
+    fn at_optional_block_hash(
```

Comment:
As above, let's put this code directly into `at()` and `at_latest()` so that the former no longer needs to be `async`, and no need for `#[inline]` :)

Reply:
Ah I see; both have to be `async` anyway actually!

In this case, let's just remove the `#[inline]` and it's all good :)